### PR TITLE
fix: throw error when SubaccountApiCredential secret is missing clien…

### LINKED ISSUE #444: [TASK] `SubaccountApiCredential` throw an error when secret contains only three keys
For a `SubaccountApiCredential` to work, it has to create a secret containing **four** keys. 

The key `client_secret` is only being returned during an initial CREATE against the BTP CLI. Subsequent GET requests fail to retrieve this key. 

This is an issue when you have an orphaned/already existing ApiCredential on BTP side, causing the  `SubaccountApiCredential` to look healthy but actually it is not, because the `client_secret` could not be returned. 

Therefore when creating a secret using the `SubaccountApiCredential` its controller should check, whether the secret to be created contains four keys and if not, raise an error message like 
`"can not read client_secret from source, please delete external resource and re-create Crossplane resource"`. 

ref
- https://github.com/SAP/crossplane-provider-btp/issues/434

### PR DESCRIPTION
**Pull request by 1122Louis**
Fixes #444

When a SubaccountApiCredential is orphaned (exists in BTP but not managed by Crossplane), the BTP API returns only 3 keys on GET,- omitting `client_secret`. This caused the resource to appear `Ready=True` with an incomplete connection secret.

This PR adds a check in `Initialize()` that verifies the connection secret contains `attribute.client_secret`. If missing, it returns an error so Crossplane marks the resource as not ready. Certificate-based credentials (where `client_secret` is never present) are skipped.

Also fixes a pre-existing nil pointer dereference when `SubaccountRef` is nil but `SubaccountID` is set.

### CODE REVIEW

**Review by sdischer-sap — CHANGES_REQUESTED** [review]
Please format the file, because right now its barely readable. 
From what I read in the description and in the test looks like you are on the right track though.

File: config/btp_subaccount_api_credential/config.go [hunk 1]
```diff
@@ -103,5 +104,21 @@ func (d *DeletionProtectionInitializer) Initialize(ctx context.Context, mg resou
 			return errors.New(providerv1alpha1.ErrResourceInUse)
 		}
 	}
+
+	if cr.Spec.ForProvider.CertificatePassed == nil {
```
Comment (sdischer-sap) [inline config/btp_subaccount_api_credential/config.go:112]: Don't really get why `CertificatePassed` should we relevant for this issue? Do you have some reasoning why only in cassed this is empty this check should apply?

File: config/btp_subaccount_api_credential/config.go [hunk 2]
```diff
@@ -103,5 +104,21 @@ func (d *DeletionProtectionInitializer) Initialize(ctx context.Context, mg resou
...
+		secretRef := cr.GetWriteConnectionSecretToReference()
+		if secretRef != nil && secretRef.Name != "" {
+			secret := &corev1.Secret{}
+			err := d.Kube.Get(ctx, client.ObjectKey{
+				Name:      secretRef.Name,
+				Namespace: secretRef.Namespace,
+			}, secret)
+			if err == nil {
+				if _, ok := secret.Data["attribute.client_secret"]; !ok {
```
Comment (sdischer-sap) [inline config/btp_subaccount_api_credential/config.go]: Did you test this end to end? 
I am not sure if there might not be some other edge cases in which a secret is generally available, but empty. If there are those cases this would prevent a creation entirely. 
I would be a bit more persimistic with narrowing the error scenario here and check that those other 3 attributes are present and only the secret is missing to make sure it really is exactly this use case.

**Review by sdischer-sap — CHANGES_REQUESTED** [review]
Great that you found this approach. 
I would change some details though to make it more robust (see remarks).

File: config/btp_subaccount_api_credential/config.go (hunk 1 above)
Comment (1122Louis) [reply config/btp_subaccount_api_credential/config.go:112]: Thanks for your comment.... According to the Terraform BTP provider docs (https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/subaccount_api_credential), client_secret is only generated "if the certificate is omitted". So when certificate_passed is set, no client_secret is ever created and the check would always fail for certificate-based credentials even when they're healthy. That's why I added the guard to skip the check in that case.

Comment (sdischer-sap) [reply config/btp_subaccount_api_credential/config.go:112]: Ah nice catch then. Then nevermind my comment, didn't know that. 
Although maybe you can add a code comment saying that according to docs this is how it works, so somebody else might not take this false assumption as me and potentially get tempted to delete that.

File: config/btp_subaccount_api_credential/config.go (hunk 2 above)
Comment (1122Louis) [reply config/btp_subaccount_api_credential/config.go]: Thank you very much, I went ahead and updated it! The check now only triggers when client_id, token_url, and api_url are all there but client_secret isn't , so it should only catch the orphaned credential case. Added a unit test too to make sure a half-filled secret doesn't cause issues.

I also tried to run e2e test locally by setting up a kind cluster with the provider installed and real BTP credentials, but ran into a 400 Bad Request from the BTP API when creating the SubaccountApiCredential, which seems to be an entitlement issue in my test subaccount, unrelated to this change. Since my fix only affects the Initialize() logic that runs when reading an existing secret, the unit tests cover the relevant scenario directly.

Let me know if you have any other feedback

File: config/btp_subaccount_api_credential/config.go (hunk 1 above)
Comment (1122Louis) [reply config/btp_subaccount_api_credential/config.go:112]: Done, added a link to the Terraform BTP provider docs in the comment

File: config/btp_subaccount_api_credential/config.go (hunk 2 above)
Comment (sdischer-sap) [reply config/btp_subaccount_api_credential/config.go]: Nice thank you. 
I added a comment here to add verification of this error behaviour as part of the import testing that is part of that ticket: https://github.com/SAP/crossplane-provider-btp/issues/553#issuecomment-4302315857. 

So no implementation work left here. However running the e2e test locally is something you will need at some point, so I would suggest to dig a bit deeper why those are failing at some point.

**Review by sdischer-sap — APPROVED** [review]
Nice work!
LGTM now

File: config/btp_subaccount_api_credential/config.go (hunk 2 above)
Comment (1122Louis) [reply config/btp_subaccount_api_credential/config.go]: Thank you for the review! I'll look into running the E2E tests locally to understand the failures better.
